### PR TITLE
Improve barman logging capabilities

### PR DESCRIPTION
**Pull request by gcalacoci**
We have found that in certain corner cases the exceptions happening during rsync backups execution could be not correctly logged and even be hidden by subsequent exceptions happening in the `finally` statement that takes care of finalizing the backup.

To avoid that, we added a `logger.debug` line that always prints in debug mode the stack trace of exception happening during the execution of rsync backups.

References: BAR-277

### CODE REVIEW

**Review by barthisrael — APPROVED** [review]
This PR looks good.

I agree with it, and I think it is better to have "double logging" in some cases than having none in some corner cases.

I performed a quick test and it seems to be working fine:

* Introduced an artificial exception before creating the copy controller:

```diff
diff --git a/barman/backup_executor.py b/barman/backup_executor.py
index 9631855..1a48716 100644
--- a/barman/backup_executor.py
+++ b/barman/backup_executor.py
@@ -1255,6 +1255,7 @@ class RsyncBackupExecutor(ExternalBackupExecutor):
             reuse_backup = self.config.reuse_backup
             safe_horizon = previous_backup.begin_time

+        raise Exception("SOMETHING WENT WRONG!")
         # Create the copy controller object, specific for rsync,
         # which will drive all the copy operations. Items to be
         # copied are added before executing the copy() method
```

* Took a backup, which failed:

```console
$ barman backup rsync
Starting backup using rsync-concurrent method for server rsync in /home/vagrant/barman_home/rsync/base/20240820T131738
Backup start at LSN: 0/E4000028 (0000000100000000000000E4, 00000028)
Starting backup copy via rsync/SSH for 20240820T131738
ERROR: The backup has failed copying files
Asking PostgreSQL server to finalize the backup.
ERROR: Backup failed writing backup label.
DETAILS: [Errno 2] No such file or directory: '/home/vagrant/barman_home/rsync/base/20240820T131738/data/backup_label'
Processing xlog segments from streaming for rsync
	0000000100000000000000E1
	0000000100000000000000E2
	0000000100000000000000E3
```

* As I had the DEBUG logs, I was able to check the real exception behind the error, i.e. the "SOMETHING WENT WRONG" exception:

```console
2024-08-20 13:17:38,688 [37561] barman.backup_executor ERROR: The backup has failed copying files
2024-08-20 13:17:38,688 [37561] barman.backup_executor DEBUG: Backup failed: SOMETHING WENT WRONG!
Traceback (most recent call last):
  File "/home/vagrant/Repositories/edb/barman/barman/backup_executor.py", line 893, in backup
    self.backup_copy(backup_info)
  File "/home/vagrant/Repositories/edb/barman/barman/backup_executor.py", line 1258, in backup_copy
    raise Exception("SOMETHING WENT WRONG!")
Exception: SOMETHING WENT WRONG!
2024-08-20 13:17:38,690 [37561] barman.backup_executor INFO: Asking PostgreSQL server to finalize the backup.
2024-08-20 13:17:38,692 [37561] barman.backup_executor DEBUG: Stop of native concurrent backup
2024-08-20 13:17:38,876 [27658] barman.command_wrappers INFO: rsync: pg_receivewal: finished segment at 0/E4000000 (timeline 1)
2024-08-20 13:17:38,876 [27656] barman.command_wrappers INFO: postgres: pg_receivewal: finished segment at 0/E4000000 (timeline 1)
2024-08-20 13:17:39,012 [37561] barman.backup_executor DEBUG: Writing backup label: /home/vagrant/barman_home/rsync/base/20240820T131738/data/backup_label
2024-08-20 13:17:39,013 [37561] barman.backup ERROR: Backup failed writing backup label.
DETAILS: [Errno 2] No such file or directory: '/home/vagrant/barman_home/rsync/base/20240820T131738/data/backup_label'
```
